kodi_addon_checker/check_addon.py

```diff
@@ -384,3 +391,33 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_path, addon_branch):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_path, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    for branch in branches:
+
+        branch_url = ROOT_URL.format(branch=branch)
+        repo_addons = _get_addons(branch_url)
```

Comment:
Now you have reintroduced the code which does 6 request for each addon in a repo. I don't care about the 6 request, but about downloading those files multiple times.

Reply:
you mean when we will run this on whole repo right ?

Reply:
you mean for whole repo check ?

Reply:
Yes for a whole repo check you this will download the same 6 files as many times as the repo has addons.
Notice: In most cases not exactly 6 x addons as the for loop has a break

Reply:
ok so maybe we can add a new function that make total of 6 reqeuests and will return a dictionary. That will have branch name as `key` and for `value` we will have another dictionary that will have respective repo_addons 
so it would look like:

```
{gotham:{<all the addon list in gotham>},'helix':{<all the addon list in helix>}}
```
Then we can use that dictionary again and again without having to make more than 6 requests